PerfCounterReporter/PerfCounterReporterService.cs

```diff
@@ -17,8 +17,42 @@ public PerfCounterReporterService()
             InitializeComponent();
         }
 
+        private void createCounterIfNotExist(string categoryName, string counterName, string counterHelp, System.Diagnostics.PerformanceCounterType type, System.Diagnostics.CounterCreationDataCollection counterDatas)
+        {
+            if(!System.Diagnostics.PerformanceCounterCategory.CounterExists(counterName, categoryName))
+            {
+                System.Diagnostics.CounterCreationData counter =
+                   new System.Diagnostics.CounterCreationData();
+                counter.CounterName = counterName;
+                counter.CounterHelp = counterHelp;
+                counter.CounterType = type;
+
+                counterDatas.Add(counter);
+            }
+        }
+
         protected override void OnStart(string[] args)
         {
+            string signalFxCategory = "SignalFX";
+
+            System.Diagnostics.CounterCreationDataCollection CounterDatas =
+                   new System.Diagnostics.CounterCreationDataCollection();
+
+            createCounterIfNotExist(signalFxCategory, "UsedMemory", "Total used memory", System.Diagnostics.PerformanceCounterType.NumberOfItems64, CounterDatas);
```

Comment:
Put this in the same class that implements the logic and call out to it from here (until we implement dynamic scanning